Assets/Scripts/Enemies/BossBehaviorController.cs

```diff
@@ -4,108 +4,156 @@
 using UnityEngine;
 using System.Linq;
 
-[RequireComponent (typeof(EnemyHealth))]
-public class BossBehaviorController : MonoBehaviour {
+public class BossAction : MonoBehaviour
+{
+    public bool actionRunning = false;
+}
+[RequireComponent(typeof(EnemyHealth))]
+public class BossBehaviorController : MonoBehaviour
+{
     [Serializable]
-    public struct Phase {
+    public struct Phase
+    {
         public float health;
-        public MonoBehaviour startBehaviour; // a behavior to enable on start
-        public MonoBehaviour endBehaviour; // a behavior to enable on end
+        public BossAction repeatingBehaviour;//  behavior that gets called everytime b4 the other possible behaviors
+        public BossAction startBehaviour; // a behavior to enable on start
+        public BossAction endBehaviour; // a behavior to enable on end
         public List<PhaseAction> backgroundActions; // behavours to be run on loop until the phase ends
-        public List<PhaseAction> PossibleActions; 
+        public List<PhaseAction> PossibleActions;
     }
     [Serializable]
-    public struct PhaseAction {
+    public struct PhaseAction
+    {
+        public bool independent;
```

Comment:
what does this do?

Reply:
since alot of the ratboss actions timings where handled by themselves in their own scripts i decided to add an option to the PhaseAction struct that lets the manager know if it should run the behavior either by the runframes variable which is handled by the bbc or the behavior themselves handle that, if the behavior themselves handle the timings then the behavior would be "independent"

Reply:
Small suggestion to add a comment in here to describe this so all the other systems will for sure know what it is?